compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -63,49 +66,121 @@ void applyConfigToPassManager(PassManager &pm, bool printIRBeforeAll,
 }
 }  // namespace
 
-void findVitis() {
-  const char *env_vitis = ::getenv("VITIS");
-  if (env_vitis == nullptr) {
-    if (auto vpp = sys::findProgramByName("v++")) {
-      SmallString<64> real_vpp;
-      std::error_code err = sys::fs::real_path(vpp.get(), real_vpp);
-      if (!err) {
-        sys::path::remove_filename(real_vpp);
-        sys::path::remove_filename(real_vpp);
-        ::setenv("VITIS", real_vpp.c_str(), 1);
-        dbgs() << "Found Vitis at " << real_vpp.c_str() << "\n";
+std::optional<std::string> findVitis(std::optional<std::string> &vitisDir) {
+  if (!vitisDir) {
+    const char *envVitis = ::getenv("VITIS");
+    if (!envVitis) {
+      if (auto vpp = sys::findProgramByName("v++")) {
+        SmallString<64> realVpp;
+        std::error_code err = sys::fs::real_path(vpp.get(), realVpp);
+        if (!err) {
+          sys::path::remove_filename(realVpp);
+          sys::path::remove_filename(realVpp);
+          vitisDir = realVpp.str().str();
+          LLVM_DEBUG(dbgs() << "Found Vitis at " << realVpp.c_str() << "\n");
+        }
       }
     }
   }
-  env_vitis = ::getenv("VITIS");
-  if (env_vitis != nullptr) {
-    SmallString<64> vitis_path(env_vitis);
-    SmallString<64> vitis_bin_path(vitis_path);
-    sys::path::append(vitis_bin_path, "bin");
-
-    SmallString<64> aietools_path(vitis_path);
-    sys::path::append(aietools_path, "aietools");
-    if (!sys::fs::exists(aietools_path)) {
-      aietools_path = vitis_path;
-      sys::path::append(aietools_path, "cardano");
+  if (vitisDir) {
```

Comment:
Replacing with `if (!vitisDir) return std::nullopt;` and then unindenting the block below would be more clean imo.

Reply:
LOL